drivers/hmis/app/models/hmis/ce/referral.rb

```diff
@@ -27,14 +27,34 @@ class Referral < GrdaWarehouseBase
     has_many :steps, class_name: 'Hmis::WorkflowExecution::Step', through: :workflow_instance, source: :steps
     has_many :current_steps, -> { preload(:node) }, class_name: 'Hmis::WorkflowExecution::Step', through: :workflow_instance, source: :open_steps
 
-    # TODO(#7395): permissions
     scope :viewable_by, ->(user) do
-      joins(:target_project).merge(Hmis::Hud::Project.viewable_by(user))
+      # What makes a referral viewable by a user?
+      # - If they have can_view_referrals at the target project, OR
+      # - If they have can_view_own_referrals, AND are assigned a step in the referral.
+
+      # Start with base scope that does all necessary joins, for structural compatibility when we `or` the scopes later
+      base_scope = joins(:target_project).left_outer_joins(:steps).left_outer_joins(steps: :assignments)
```

Comment:
I think the sql is equivalent if you remove `left_outer_joins(:steps)` since its joined to get to `assignments` anyway